substra/sdk/schemas.py

```diff
@@ -130,6 +130,65 @@ def build_request_kwargs(self, local):
             yield data, None
 
 
+class ComputePlanTraintupleSpec(_Spec):
+    algo_key: str
+    data_manager_key: str
+    train_data_sample_keys: List[str]
+    traintuple_id: str
+    in_models_ids: Optional[List[str]]
+    tag: Optional[str]
+    metadata: Optional[Dict[str, str]]
+
+
+class ComputePlanAggregatetupleSpec(_Spec):
+    aggregatetuple_id: str
+    algo_key: str
+    worker: str
+    in_models_ids: Optional[List[str]]
+    tag: Optional[str]
+    metadata: Optional[Dict[str, str]]
+
+
+class ComputePlanCompositeTraintupleSpec(_Spec):
+    composite_traintuple_id: str
+    algo_key: str
+    data_manager_key: str
+    train_data_sample_keys: List[str]
+    in_head_model_id: Optional[str]
+    in_trunk_model_id: Optional[str]
+    tag: Optional[str]
+    out_trunk_model_permissions: Permissions
+    metadata: Optional[Dict[str, str]]
+
+
+class ComputePlanTesttupleSpec(_Spec):
+    objective_key: str
+    traintuple_id: str
+    tag: Optional[str]
+    data_manager_key: Optional[str]
+    test_data_sample_keys: Optional[List[str]]
+    metadata: Optional[Dict[str, str]]
+
+
+class _BaseComputePlanSpec(_Spec, abc.ABC):
+    traintuples: Optional[List[ComputePlanTraintupleSpec]]
+    composite_traintuples: Optional[List[ComputePlanCompositeTraintupleSpec]]
+    aggregatetuples: Optional[List[ComputePlanAggregatetupleSpec]]
+    testtuples: Optional[List[ComputePlanTesttupleSpec]]
+
+
+class ComputePlanSpec(_BaseComputePlanSpec):
+    tag: Optional[str]
+    clean_models: Optional[bool]
+    metadata: Optional[Dict[str, str]]
+
+    type_: typing.ClassVar[Type] = Type.ComputePlan
+
+
+class UpdateComputePlanSpec(_BaseComputePlanSpec):
+    pass
```

Comment:
Aren't you missing the `type_` here?

Reply:
in the Type class, there is no type 'updateComputePlan', I thought that it was only for the types of stored objects

Reply:
Ok, having a type on this object is not strictly necessary, let's not do it then :)